src/request.ts

```diff
@@ -55,18 +58,35 @@ export const getDefaultAgent = (url: URL) => {
     : new HttpProxyAgent(proxy, httpAgentOptions);
 };
 
-export const buildHeaders = (
-  appName?: string,
-  instanceId?: string,
-  etag?: string,
-  contentType?: string,
-  custom?: CustomHeaders,
-  specVersionSupported?: string,
-): Record<string, string> => {
+type HeaderOptions = {
+  appName?: string;
+  instanceId?: string;
+  etag?: string;
+  contentType?: string;
+  custom?: CustomHeaders;
+  specVersionSupported?: string;
+  connectionId?: string;
+};
+
+export const buildHeaders = ({
+  appName,
+  instanceId,
+  etag,
+  contentType,
+  custom,
+  specVersionSupported,
+  connectionId,
+}: HeaderOptions): Record<string, string> => {
   const head: Record<string, string> = {};
   if (appName) {
+    // TODO: delete
     head['UNLEASH-APPNAME'] = appName;
+    // TODO: delete
```

Comment:
delete when, though? Next major?

Reply:
I want to have a super safe version upgrade that I can recommend to everyone even if for some unknown reason they relied on he old headers